openedx/core/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -3,23 +3,60 @@
 """
 
 import json
+from datetime import datetime
 
 import ddt
 from django.conf import settings
 from django.test import RequestFactory, TestCase
 from django.core.urlresolvers import reverse
 import httpretty
+from oauth2_provider import models as dot_models
 from provider import constants
+from pytz import utc
 import unittest
 
 from student.tests.factories import UserFactory
 from third_party_auth.tests.utils import ThirdPartyOAuthTestMixin, ThirdPartyOAuthTestMixinGoogle
 
 from .constants import DUMMY_REDIRECT_URL
+from . import mixins
 from .. import adapters
+from .. import models
+
 if settings.FEATURES.get("ENABLE_OAUTH2_PROVIDER"):
     from .. import views
-from . import mixins
+
+
+class AccessTokenLoginMixin(object):
+    """
+    Shared helper class to assert proper access levels when using access_tokens
+    """
+
+    login_with_access_token_url = reverse("login_with_access_token")
```

Comment:
Defining this as a class variable won't work in CMS tests since "login_with_access_token" is not defined when `ENABLE_OAUTH2_PROVIDER` is unset.  You can set it in the `setUp` method (you'll need to correct the Mixin class order - add the Mixin before the TestCase class).

This is what is causing the CMS test failures:
https://build.testeng.edx.org/job/edx-platform-test-subset/617147/console